src/js/models/covid19/budgetCategories/BaseBudgetCategoryRow.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * BaseBudgetCategoryRow.js
+ * Created by James Lee 6/05/20
+ */
+
+import { formatMoney } from 'helpers/moneyFormatter';
+import { budgetColumns, budgetColumnFields, budgetDropdownColumns, budgetFields } from 'dataMapping/covid19/budgetCategories/BudgetCategoriesTableColumns';
+
+const BaseBudgetCategoryRow = {
+    populateOnSpendingCategory(data, spendingCategory) {
+        budgetDropdownColumns[spendingCategory].forEach((column) => {
+            const defaultValue = typeof (data[budgetFields[spendingCategory][column.title]]) === 'string' ? '' : 0;
+            this[`_${column.title}`] = data[budgetFields[spendingCategory][column.title]] || defaultValue;
+        });
+    },
+    populateBase(data, type) {
+        let defaultValue = null;
+        budgetColumns[type].forEach((column) => {
+            if (type === 'def_code') {
+                if (budgetColumnFields[column.title]) {
+                    defaultValue = typeof (data[budgetColumnFields[column.title]]) === 'string' ? '' : 0;
+                    this[`${column.title}`] = data[budgetColumnFields[column.title]] || defaultValue;
+                }
+            } else if (column.title === 'name') {
+                this[`${column.title}`] = `${data.code} — ${data.description}` || '';
+            } else {
+                defaultValue = typeof (data[column.title]) === 'string' ? '' : 0;
+                this[`${column.title}`] = data[column.title] || defaultValue;
+            }
+        });
+    },
+    populateTotalBudgetaryResources(data, type, spendingCategory) {
+        if (type !== 'object_class' && spendingCategory !== 'award_spending') {
```

Comment:
This is confusing to me. Why are we only seeing the total based on type?

Reply:
We're only supposed to show data for `total budgetary resources` if the `spendingCategory` is `total_spending` and when the `type` is anything other than `object_class`.